.github/workflows/bump-version-tag.yml

```diff
@@ -37,24 +59,33 @@ jobs:
         run: |
           CURRENT_VERSION="$(grep -E '__version__' finch/__version__.py | cut -d ' ' -f3)"
           echo "current_version=${CURRENT_VERSION}"
+          echo "CURRENT_VERSION=${CURRENT_VERSION}" >> $GITHUB_ENV
       - name: Bump Version
         if: ${{ github.event.inputs.tag == 'false' }}
         run: |
           pip install bump-my-version
           echo "running `bump-my-version bump  --no-tag ${{ github.event.inputs.bumpversion }}`"
           NEW_VERSION="$(grep -E '__version__' finch/__version__.py | cut -d ' ' -f3)"
           echo "new_version=${NEW_VERSION}"
-      - name: Bump Version with Tagging
-        if: ${{ github.event.inputs.tag == 'true' }}
+      - name: Install bump-my-version
         run: |
-          pip install bump-my-version
-          echo "running `bump-my-version bump ${{ github.event.inputs.bumpversion }}`"
-          NEW_VERSION="$(grep -E '__version__' finch/__version__.py | cut -d ' ' -f3)"
+          python -m pip install bump-my-version>=0.18.3
+      - name: Conditional Bump
+        id: bump
+        run: |
+          if [[ ${{ env.CURRENT_VERSION }} =~ -dev(\.\d+)? ]]; then
+            echo "Development version (ends in 'dev(\.\d+)?'), bumping 'build' version"
+            bump-my-version bump build
+          else
+            echo "Version is stable, bumping 'patch' version"
+            bump-my-version bump patch
```

Comment:
Version bump automatically on each merge to `master`?  That's agressive !

Reply:
It bumps the version to `vX.Y.Z-dev.#` to indicate that the master branch is different from the last release or the last PR merge. We don't release until we perform a `bump-my-version bump release` to drop the `-dev`.

Reply:
Ah ok, so each merge do not auto create a release, good then.